### ISSUE TITLE
system_cmds project leads to segmentation fault

### ISSUE THREAD

**Issue by matthewbauer** [facebook/xcbuild#188]
I'm getting some weird issues with "system_cmds" (available at https://opensource.apple.com/source/system_cmds/system_cmds-671.10.3/):

```
running xcodebuild
Build settings from the command line:
    DERIVED_DATA_DIR=.
    OTHER_CFLAGS=-F/nix/store/ndf3c44581qw8pb32w4p4ckmnqdc2cab-CF-osx-10.9.5/Library/Frameworks -isystem /nix/store/b7v7sddx3dfpkycpjgk0p6fnsyhsyqdv-libc++-3.7.1/include
    OTHER_LDFLAGS=-rpath /nix/store/cjmvbdgi52qh657vcn3ry9yl8gswx5h7-system_cmds-671.10.3/lib  -L/nix/store/b7v7sddx3dfpkycpjgk0p6fnsyhsyqdv-libc++-3.7.1/lib
    SDKROOT=macosx10.10

Segmentation fault: 11
```

This may be because of some missing specs. If anyone has a working Xcode setup, it would be helpful if they can verify that it works on their machines.

**Comment by grp** [comment]
Taking a look. Debugger says to be a recursion during target dependency resolution, probably a loop in the graph. Turned on logging in `DependencyResolver.cpp`, seems like the target `login` is being found to have an implicit dependency on itself. Looking at why...

**Comment by grp** [comment]
Interesting: it's because the `login` target copies a file called `login` during its `Copy Files` build phase, and xcbuild tracks implicit dependencies by name. Many projects have implicit dependencies using wildly incorrect identifiers, so name seems important, but maybe it should have to match the product file in more than just name. File type? Or maybe it just count as an implicit dependency, and just break the cycle if one is detected?

**Comment by copumpkin** [comment]
Any further thoughts on this one?

**Comment by grp** [comment]
Not really. I don't know what the correct behavior should be so hard to start to implement it. You could patch the project file to rename the copied `login` file or the `login` target for now, perhaps.

**Comment by shlevy** [comment]
@grp How can we rename the target? I just changed `name = login` to `name = login_proj` insidethe `PBXNativeTarget` for `login` and still get a segfault.

**Comment by eraserhd** [comment]
Is this the same issue trying to build [Hammerspoon](https://github.com/Hammerspoon/hammerspoon)?:

```
[nix-shell:~/src/hammerspoon]$ xcodebuild -workspace Hammerspoon.xcworkspace -scheme Release -target Hammerspoon
warning: unhandled ShellScriptBuildPhase key inputFileListPaths
warning: unhandled ShellScriptBuildPhase key outputFileListPaths
warning: unhandled FileReference key indentWidth
warning: unhandled FileReference key tabWidth
Build settings from the command line:
    DERIVED_DATA_DIR=.
    SDKROOT=MacOSX10.12

Segmentation fault: 11

```
